src/ug-core/conversation-manager/ConversationManager.ts

```diff
@@ -209,7 +209,72 @@ export class ConversationManager extends EventEmitter implements IConversationMa
     await this.setState('idle')
   }
 
+  /**
+   * Dispose of all resources and clean up.
+   * Should be called when the conversation ends or when navigating away.
+   * This will:
+   * - Stop audio recording and release media stream
+   * - Stop VAD analysis
+   * - Dispose audio recorder
+   * - Dispose playback manager (closes AudioContext)
+   * - Close WebSocket connection
+   * - Remove all event listeners from this and child components
+   */
+  async dispose(): Promise<void> {
+    if (this.disposed) {
+      this.logger.warn('ConversationManager already disposed')
+      return
+    }
+    
+    this.logger.debug('Disposing ConversationManager...')
+    this.disposed = true
+    
+    // Remove event listeners from child components BEFORE disposing them
+    // This prevents handlers from firing after disposal
+    this.userInputManager?.removeAllListeners()
+    this.playbackManager?.removeAllListeners()
+    this.network?.removeAllListeners()
+    
+    // Set state to 'idle' directly (bypass setState since we're disposed)
+    this.state = 'idle'
+    
+    try {
+      
+      // Dispose user input manager (stops recording, VAD, and releases media stream)
+      if (this.userInputManager && typeof (this.userInputManager as any).dispose === 'function') {
+        (this.userInputManager as any).dispose()
+      } else {
+        // Fallback: stop user input and release media stream manually
+        await this.userInputManager.stop()
+        if (this.mediaStream) {
+          this.mediaStream.getTracks().forEach((track) => {
+            track.stop()
+            this.logger.debug(`Stopped media track: ${track.kind}`)
+          })
+          this.mediaStream = null
+        }
+      }
+      
+      // Dispose playback manager (will destroy audio player and close AudioContext)
+      if (this.playbackManager && typeof this.playbackManager.dispose === 'function') {
+        this.playbackManager.dispose()
+      }
+      
+      // Disconnect network (closes WebSocket)
+      await this.network.disconnect()
+      
+      // Remove all event listeners from this manager
+      this.removeAllListeners()
+      
+      this.logger.debug('ConversationManager disposed successfully')
+    } catch (error) {
+      this.logger.error('Error during disposal', error)
+      throw error
```

Comment:
Im not sure in the case we cant dispose - we should throw anything. I think writing to log should be enough